File: 01-frontend/zarzadzanie-magazynem/src/app/common/material/material.ts
import { Category } from "../category/category";
import { Vendor } from "../vendor/vendor";

export class Material {
  materialId: number;
  materialNumber: string;
  materialManufacturer: string;
  materialName: string;
  materialPrice: number;
  materialQuantity: number;
  unitOfMeasure: string;
  materialEAN: string;
  materialCreated: Date;
  materialSafetyStock: number;
  materialDescription: string;
  materialCategory: Category;
  materialVendor: Vendor;
  materialStatus: boolean;

  constructor() { }

  static fromJSON(json: any): Material {
    const material = new Material();
    Object.assign(material, json);
    return material;
  }
}

File: 02-backend/zarzadanie-magazynem/src/main/java/com/kc6379/zarzadaniemagazynem/model/Material.java
package com.kc6379.zarzadaniemagazynem.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@Entity
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Material {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long materialId;
    private String materialNumber;
    private String materialManufacturer;
    private String materialName;
    private Double materialPrice;
    private Integer materialQuantity;
    private String unitOfMeasure;
    private String materialEAN;
    private Instant materialCreated;
    private Integer materialSafetyStock;
    private String materialDescription;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(
            name = "categoryId",
            referencedColumnName = "categoryId"
    )
    private Category materialCategory;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(
            name = "vendorId",
            referencedColumnName = "vendorId"
    )
    private Vendor materialVendor;
    private Boolean materialStatus;
}
